js/repeater.js

```diff
@@ -699,6 +699,13 @@
 			var height;
 			var viewportMargins;
 
+			var $hiddenParents = this.$element.parents().addBack().not(':visible');
```

Comment:
also limit the set to first `:visible` ancestor. this will minimize the number of parents visited.